common/flatpak-dir.c

```diff
@@ -2326,6 +2326,14 @@ repo_pull_one_dir (OstreeRepo          *self,
       g_variant_builder_add (&find_builder, "{s@v}", "update-frequency",
                              g_variant_new_variant (g_variant_new_uint32 (update_freq)));
 
+      if (flatpak_flags & FLATPAK_PULL_FLAGS_ALLOW_DOWNGRADE)
+        {
+          revs_to_fetch[0] = rev_to_fetch;
+          revs_to_fetch[1] = NULL;
+          g_variant_builder_add (&find_builder, "{s@v}", "override-commit-ids",
+                                 g_variant_new_variant (g_variant_new_strv ((const char * const *) revs_to_fetch, -1)));
```

Comment:
You can simplify the strv handling:
```
g_variant_new_strv (&rev_to_fetch, 1)
```

Also, are you sure you need the `g_variant_new_variant()` call there? I thought the `@v` format string automatically wrapped things in a variant. I’m not 100% sure, though, so you should check.

Reply:
I couldn't tell from the docs, so I'll leave it the way the rest of the code does it.